packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/LinkToCategoriesTests.java

```diff
@@ -145,6 +145,75 @@ public void testLinkContentToMultipleCategories()
         fileNode.assertThat().field(PROPERTIES_FIELD).contains(secondCategory.getId());
     }
 
+    /**
+     * Try to link file to three categories, two of which are the same, and expect two distinct categories in output.
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_withRepeatedCategory()
+    {
+        STEP("Check if file is not linked to any category");
+        RestNodeModel fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+
+        fileNode.assertThat().field(ASPECTS_FIELD).notContains("cm:generalclassifiable");
+        fileNode.assertThat().field(PROPERTIES_FIELD).notContains("cm:categories");
+
+        STEP("Link content to three (one repeated) categories and expect 201");
+        final RestCategoryModel secondCategory = prepareCategoryUnderRoot();
+        final List<RestCategoryLinkBodyModel> categoryLinkModels = List.of(
+            createCategoryLinkModelWithId(category.getId()),
+            createCategoryLinkModelWithId(secondCategory.getId()),
+            createCategoryLinkModelWithId(category.getId())
+        );
+        final RestCategoryModelsCollection linkedCategories = restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategories(categoryLinkModels);
+
+        restClient.assertStatusCodeIs(CREATED);
+        linkedCategories.assertThat().entriesListCountIs(2);
+        linkedCategories.getEntries().get(0).onModel().assertThat().isEqualTo(category);
+        linkedCategories.getEntries().get(1).onModel().assertThat().isEqualTo(secondCategory);
+
+        STEP("Verify if repeated category was ignored and only two categories are present in file metadata");
+        fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+
+        fileNode.assertThat().field(PROPERTIES_FIELD).containsOnce(category.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).containsOnce(secondCategory.getId());
```

Comment:
I think we really want to check that there is a property `cm:categories` with exactly the set of values `{categoryId1, categoryId2}`